scripts/install-darwin-multi-user.sh

```diff
@@ -222,3 +222,35 @@ EOF
         failure "This script needs a /nix volume with global permissions! This may require running sudo /usr/sbin/diskutil enableOwnership /nix."
     fi
 }
+
+# appended to shell init
+poly_extra_init_for_zshenv() {
+    cat <<'EOF'
+
+# wrap path_helper to keep it from pushing nix rightwards
+/usr/libexec/path_helper() {
+    case "$1" in
+        "-c")
+            # don't interfere if this gets invoked with -c
+            command /usr/libexec/path_helper "$@"
+            ;;
+        *)
+            (
+                # subshell to see what path_helper would do to PATH. inject a
+                # separator for splitting (saving previous PATH can duplicate
+                # entries, and people tend to see that as a smell)
+                eval "$(PATH="nixpathsep:$PATH" command /usr/libexec/path_helper "$@")"
+                local prefixed="${PATH%%:nixpathsep:*}" pushed_back="${PATH##*:nixpathsep:}"
+
+                # re-emit what path_helper would; remove function to avoid rerunning
+                echo "PATH=\"${pushed_back}:${prefixed}\"; export PATH; unset -f /usr/libexec/path_helper;"
+            )
+            ;;
+    esac
+
+    # most-conservative conceivable implementation--only run once
+    unset -f /usr/libexec/path_helper
```

Comment:
The [gist you shared](https://gist.github.com/abathur/877a4fa25ec7047baeb3033a9153b269) does not contain this line. I've built your branch from source now and copied the exact contents from this commit to `/etc/zshenv` and tried everything again, including login via ssh.

Everything still works flawlessly, even the special case of a flake that it itself installs nix:

```
$ whence -a nix
/Users/feuh/.nix-profile/bin/nix
/nix/var/nix/profiles/default/bin/nix
$ nix --version
nix (Nix) 2.15.1
$ nix develop -c zsh
$ whence -a nix
/nix/store/ibzj90n2lxmd8jbph5izcgz76f46r514-nix-2.17.1/bin/nix
/Users/feuh/.nix-profile/bin/nix
/nix/var/nix/profiles/default/bin/nix
$ nix --version
nix (Nix) 2.17.1
```

This works in both kitty and Terminal, and I tried replacing `nix develop -c zsh` with `direnv allow`, and the impact on the path is identical, which is great!

Reply:
Good catch. Sorry--it's vestigial. I'll remove it in a force push later. (This one won't have an effect since it's inside the subshell; I moved it into the text echoed for eval and forgot to clean up when I confirmed it.)

Reply:
I've force-pushed to remove this. I'll update all of the links/paths once CI finishes. I'll leave the conversation unresolved to avoid hiding the feedback though. :)

Reply:
Great! I moved that feedback into my main review comment, so you can mark this as resolved to keep things tidy